install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -2708,6 +2708,9 @@ clustermesh:
     #     cert: ""
     #     key: ""
     #     caCert: ""
+    # -- The maximum number of clusters to support in a clustermesh.
+    # Values > 255 will decrease maximum allocatable identities per cluster.
+    maxConnectedClusters: 255
```

Comment:
Can this value be 256? Can it be 257? Can it be 510? Can it be 517? It's a hint that maybe we should have some helm validation in this field :smile:

Reply:
Two aspects in addition to André's comment here:

* I'd explicitly state which are the supported values;
* How would this play with the `config.enabled` flag above? More in general, I wonder if is the correct place for this setting, as `config` here currently represents the list of remote clusters. For instance, the cluster ID and cluster name settings, which are also clustermesh-related, are in the global agent section. Should this also be there? Should we even not expose this through helm, and make it configurable only through the `extraConfig` section? This option might make sense if we expect this to be used only in very niche subset of scenarios, and if properly documented.

Reply:
good point on the config aspect. I moved `maxConnectedClusters` to `clustermesh.maxConnectedClusters` and added helm validation.

@giorio94 I think your suggestion about extraConfig also makes sense, given that this might not be a widely used feature. If either of you think this is a better fit, I can look at making that change